src/login/logind-button.c

```diff
@@ -495,6 +634,9 @@ int button_open(Button *b) {
 
         (void) button_set_mask(b->name, fd);
 
+        if (!button_get_modifier_mask(b))
+                return log_warning_errno(errno, "Failed to get modifier mask on %s: %m", p);
+
```

Comment:
hmm, you use LOG_WARNING level here, but make this fatal. If you make things fatal (i.e. return with an error), then you should log at LOG_ERR level, i.e. log_error_errno()